From 5a1f9d261b406f5f33b150c9d9c3d3e2c8d779db Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Wed, 18 Oct 2017 14:51:50 +0100
Subject: [PATCH] lib: Address upstream util-linux partlist regression

This fixes a segfault occuring within clr-boot-manager's usage of the
`blkid_partlist_numof_partitions` function that will now segfault when
passed a NULL blkid_partlist, as of util-linux 2.30.x series.

This change will ensure all consumers of the cbm_blkid API will continue to
function as before, but safe guard against the util-linux internal changes
that broke the counter function to determine how many partitions are present.

Note that this behaviour only manifested on LVM installations, which have
a more advanced probing scheme within clr-boot-manager.

Solus Issue: https://dev.solus-project.com/T4763

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/lib/blkid_stub.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/lib/blkid_stub.c b/src/lib/blkid_stub.c
index 6f34ea8..794015b 100644
--- a/src/lib/blkid_stub.c
+++ b/src/lib/blkid_stub.c
@@ -151,6 +151,9 @@ blkid_partlist cbm_blkid_probe_get_partitions(blkid_probe pr)
 
 int cbm_blkid_partlist_numof_partitions(blkid_partlist ls)
 {
+        if (!ls) {
+                return 0;
+        }
         return blkid_ops->partlist_numof_partitions(ls);
 }
 
-- 
2.14.2

